### PR TITLE
Implement RemovePath and pebble rm command

### PR DESCRIPTION
**Pull request by anpep**

### CODE REVIEW

File: client/files.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	"encoding/json"
+	"fmt"
+)
+
+// RemovePathOptions holds the options for a call to RemovePath.
+type RemovePathOptions struct {
+	// Path is the absolute path to be deleted (required).
+	Path string
+
+	// MakeParents, if true, specifies that any non-existent parent directories
```
Comment (benhoyt) [inline client/files.go:28]: Comment needs updating.

File: client/files.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+type fileResult struct {
+	Path  string       `json:"path"`
+	Error *errorResult `json:"error,omitempty"`
+}
+
+type errorResult struct {
+	Message string      `json:"message"`
+	Kind    string      `json:"kind,omitempty"`
+	Value   interface{} `json:"value,omitempty"`
+}
```
Comment (benhoyt) [inline client/files.go:54]: There's going to be quite a lot of merge conflicts when we merge all of these. :-) But that's fine -- I think `fileResult` and `errorResult` should be general across all the file APIs.

File: client/files.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+// RemovePath deletes files and directories.
```
Comment (benhoyt) [inline client/files.go:56]: What about "deletes a file or directory", as it only deletes one (possibly recursively).

File: client/files.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+// The error returned is a *Error if the request went through successfully
+// but there was an OS-level error creating the directory, with the Kind
```
Comment (benhoyt) [inline client/files.go:58]: This comment needs updating.

File: cmd/pebble/cmd_rm.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	Recursive bool `short:"r" long:"recursive"`
+
+	Positional struct {
+		Path string `positional-arg-name:"<path>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var rmDescs = map[string]string{
+	"recursive": "Create parent directories as needed.",
```
Comment (benhoyt) [inline cmd/pebble/cmd_rm.go:34]: Text needs updating.

File: cmd/pebble/cmd_rm.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+var shortRmHelp = "Remove files and directories"
+var longRmHelp = `
+The rm command removes a file or directory.
+If --recursive is specified, all files and directories contained
+within the specified path will also be removed.
```
Comment (benhoyt) [inline cmd/pebble/cmd_rm.go:41]: IMO the second sentence can be removed, as it should be documented under the `--recursive` argument's help.

**Review by benhoyt — COMMENTED** [review]
Wasn't sure whether this was ready for review yet, but I did a quick pass. I'll review more thoroughly later -- just let me know.

File: client/files.go (hunk 2 above)
Comment (anpep) [reply client/files.go:54]: You're completely right (: I didn't find any alternative to this while working on multiple related functionalities at once, and didn't want any PR to be dependent on each other.

File: client/files.go (hunk 3 above)
Comment (anpep) [reply client/files.go:56]: That's slightly better, thank you!

File: cmd/pebble/cmd_rm.go (hunk 6 above)
Comment (anpep) [reply cmd/pebble/cmd_rm.go:41]: I agree with you. Here I mimicked the help text of other commands such as `add` which sometimes include a sentence about an option that's relevant to the command.

https://github.com/canonical/pebble/blob/b13464b97acd35b140b1ad2758ce745f0735daf8/cmd/pebble/cmd_add.go#L40-L45

File: client/files.go (hunk 2 above)
Comment (benhoyt) [reply client/files.go:52]: Yeah, I think that's the right choice for review -- thanks!

File: client/files.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	Path      string `json:"path"`
+	Recursive bool   `json:"recursive"`
+}
+
+type fileResult struct {
+	Path  string       `json:"path"`
+	Error *errorResult `json:"error,omitempty"`
+}
+
+type errorResult struct {
```
Comment (benhoyt) [inline client/files.go:48]: Thinking about this now, we should be able to JSON-decode this directly to `*client.Error`, which already has these fields and the JSON tags. Similar for the other PRs.

File: cmd/pebble/cmd_rm.go [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"github.com/jessevdk/go-flags"
+
+	"github.com/canonical/pebble/client"
+)
+
+type cmdRm struct {
+	clientMixin
+
+	Recursive bool `short:"r" long:"recursive"`
```
Comment (benhoyt) [inline cmd/pebble/cmd_rm.go:26]: Similar here as other PRs: let's remove `--recursive` and just have `-r`.

**Review by benhoyt — COMMENTED** [review]
We skimmed through this with Gustavo during our London mini-sprint, and it looks good. He's going to take one more pass before we merge.

**Review by niemeyer — APPROVED** [review]
This looks good, thanks Angel.  Looks ready for merging, just pending conflict resolution with the prior list files PR.

**Pull request merged**